app/src/main/java/com/gh4a/activities/RepositoryActivity.java

```diff
@@ -374,19 +327,75 @@ protected Intent navigateUp() {
         return UserActivity.makeIntent(this, mRepoOwner);
     }
 
+    private void loadWatchingStatus(boolean refresh) {
+        RepositoryService.loadWatchingStatus(this, mRepoOwner, mRepoName, refresh)
+                .subscribe(result -> {
+                    mIsWatching = (Boolean) result;
```

Comment:
Remove all no longer needed casts, give proper names instead of `result`. _Applies to other similar methods too._